pkg/polaris/polaris_test.go

```diff
@@ -983,3 +987,577 @@ func TestGcpPermissions(t *testing.T) {
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, ns := range nss {
+		fmt.Printf("%v\n", ns)
+		// all returned values should match the filter
+		if ns.K8sClusterID != testClusterID {
+			t.Fatalf(
+				"failed to filter based on clusterID: found clusterID %s, but expected %s",
+				ns.K8sClusterID,
+				testClusterID,
+			)
+		}
+	}
+
+}
+
+func TestUnassignSLA(t *testing.T) {
+	//if !boolEnvSet("TEST_INTEGRATION") {
+	//	t.Skipf("skipping due to env TEST_INTEGRATION not set")
+	//}
+
+	testNSID := uuid.MustParse("3b3d22b7-385c-5865-bd8f-0ff7534db42b")
+	ctx := context.Background()
+	testServiceAccount := ServiceAccount{
+		ClientID:       "client|YVaYtseZQXRSiwEHV8HaRKfiRfsU0BhX",
+		ClientSecret:   "BS0-2olXRhY51R1AW0qj9gdgYBOs6x4uJbUg-7DXIxQImigEiavo819R0ZTPwq8a",
+		Name:           "np-test",
+		AccessTokenURI: "https://demo.dev-017.my.rubrik-lab.com/api/client_token",
+	}
+	client, err := NewClient(ctx, &testServiceAccount, &polaris_log.DiscardLogger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	oks, err := client.Core().AssignSlaForSnappableHierarchies(
+		ctx,
+		nil,
+		core.NoAssignment,
+		[]uuid.UUID{testNSID},
+		[]core.SnappableLevelHierarchyType{},
+		true,  // shouldApplyToExistingSnapshots
+		false, // shouldApplyToNonPolicySnapshots
+		core.RetainSnapshots,
+	)
+	// Load configuration and create client. Usually resolved using the
+	// environment variable RUBRIK_POLARIS_SERVICEACCOUNT_FILE.
+	//polAccount, err := DefaultServiceAccount(true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("%v\n", oks)
+
+}
+
+// TestTakeK8NamespaceSnapshot verifies that the SDK can take a
+// namespace snapshot
+func TestTakeK8NamespaceSnapshot(t *testing.T) {
+	testClusterID := uuid.MustParse("a840e205-ac36-408d-9a12-7690769aaa88")
+	ctx := context.Background()
+	testServiceAccount := ServiceAccount{
+		ClientID:       "client|2YDsG5FIYWQ7OW8AiRqA2tPwQEwhHpfU",
+		ClientSecret:   "ZLeoEHA6dvDGciAO1bFIIWJmdjlLLawFa6WC8IdpBfgphecsUQYB4pAOKovQbt4e",
+		Name:           "kupatest",
+		AccessTokenURI: "https://demo.dev-017.my.rubrik-lab.com/api/client_token",
+	}
+
+	client, err := NewClient(ctx, &testServiceAccount, &polaris_log.DiscardLogger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	nss, err := client.K8s().ListK8sNamespace(ctx, testClusterID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(nss) == 0 {
+		t.Fatal("No namespace found")
+	}
+
+	ns := nss[0] // get first namespace
+	info, err := client.K8s().TakeK8NamespaceSnapshot(ctx, ns.ID, "00000000-0000-0000-0000-000000000000")
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Printf("%v\n", info)
+}
+
+// TestGetTaskchainInfo verifies that the SDK can fetch taskchain states
+func TestGetTaskchainInfo(t *testing.T) {
+	testClusterID := uuid.MustParse("a840e205-ac36-408d-9a12-7690769aaa88")
+	ctx := context.Background()
+	testServiceAccount := ServiceAccount{
+		ClientID:       "client|2YDsG5FIYWQ7OW8AiRqA2tPwQEwhHpfU",
+		ClientSecret:   "ZLeoEHA6dvDGciAO1bFIIWJmdjlLLawFa6WC8IdpBfgphecsUQYB4pAOKovQbt4e",
```

Comment:
:exclamation:**Cycode**: Secret of type: 'Base64 Entropy' was found.
Severity: Medium <!-- 867414fc-e8ee-4aa0-810c-48af12407a06 -->